report/verification/mms/simple_transient.md

```diff
@@ -0,0 +1,329 @@
+Moreover, a Dirichlet boundary condition will be assumed on the boundaries $\partial \Omega $.
+
+The problem is therefore:
+
+$$
+\begin{align}
+    &\nabla \cdot (D \ \nabla{c}) - \frac{\partial c}{\partial t} = -S  \quad \text{on }  \Omega  ; \ t\geq 0 \\
+    & c = c_0 \quad \text{on }  \partial \Omega ; \ t\geq 0 \\
+    & c = c_\mathrm{initial} \quad \text{on } \partial \Omega ; \ \text{at } t=0
+\end{align}
+$$(problem_simple_transient)
+
+The exact solution for mobile concentration is:
+
+$$
+\begin{equation}
+    c_\mathrm{exact} = 1 + 2 x^2 + 3 y^2 t + 2t
+\end{equation}
+$$(c_exact_simple_transient)
+
+```{note}
+We use a manufactured solution that varies linearly with time ($t^1$), as the backward Euler scheme provides an exact solution in this case.
+```
+
+Injecting {eq}`c_exact_simple_transient` in {eq}`problem_simple_transient`, we obtain the expressions of $S$, $c_0$, and $c_\mathrm{initial}$:
+
+\begin{align}
+    & S = 2 + 3 y^2 - (4 + 6t) D \\
+    & c_0 = c_\mathrm{exact} \\
+    & c_\mathrm{initial} = c_\mathrm{exact}(t=0)
+\end{align}
+
+We can then run a FESTIM model with these values and compare the numerical solution with $c_\mathrm{exact}$.
+
++++
+
+## FESTIM code
+
+```{code-cell} ipython3
+:tags: [hide-cell]
+
+from mpi4py import MPI
+import festim as F
+from dolfinx.mesh import create_unit_square
+import numpy as np
+
+fenics_mesh = create_unit_square(MPI.COMM_WORLD, 100, 100)
+
+boundary = F.SurfaceSubdomain(id=1)
+H = F.Species("mobile", mobile=True)
+D = 2
+my_mat = F.Material(D_0=D, E_D=0)
+volume = F.VolumeSubdomain(id=1, material=my_mat)
+boundary = F.SurfaceSubdomain(id=1)
+
+exact_solution = (
+    lambda x, t: 1 + 2 * x[0] ** 2 + 3 * t * x[1] ** 2 + 2 * t
+)  # exact solution
+
+f = lambda x, t: 2 + 3 * x[1] ** 2 - (4 + 6 * t) * D
+
+final_time = 17
+slices = 4
+slice_size = final_time / slices
+my_milestones = list(np.linspace(slice_size, final_time, slices))
+
+
+my_subdomains = [volume, boundary]
+my_species = [H]
+my_sources = [F.ParticleSource(value=f, volume=volume, species=H)]
+my_boundary_conditions = [
+    F.FixedConcentrationBC(subdomain=boundary, value=exact_solution, species=H)
+]
+my_settings = F.Settings(
+    atol=1e-10,
+    rtol=1e-10,
+    transient=True,
+    final_time=final_time,
+    stepsize=F.Stepsize(
+        initial_value=0.25,
+        growth_factor=1.0,
+        target_nb_iterations=30,
+        milestones=my_milestones,
+    ),
+)
+
+my_model = F.HydrogenTransportProblem(
+    mesh=F.Mesh(fenics_mesh),
+    subdomains=my_subdomains,
+    species=my_species,
+    sources=my_sources,
+    boundary_conditions=my_boundary_conditions,
+    temperature=500,
+    settings=my_settings,
+    exports=[
+        F.VTXSpeciesExport(
+            field=H,
+            filename="simple_transient_mobile.bp",
+            subdomain=volume,
+            checkpoint=True,
+        )
+    ],
+)
+
+my_model.initialise()
+my_model.run()
+```
+
+## Comparison with exact solution
```

Comment:
It's not clear without digging in the cell content what time this is.

Since you have functions to do everything, could you not plot several times too?